velox/dwio/parquet/reader/ParquetReader.cpp

```diff
@@ -782,6 +782,10 @@ size_t ParquetReader::numberOfRowGroups() const {
   return readerBase_->fileMetaData().row_groups.size();
 }
 
+const thrift::FileMetaData& ParquetReader::getFileMetaData() const {
```

Comment:
It's not a best practice to return thrift structures as that introduces a dependency on thrift at the call site.
I see that the DwrfReader has the following API. Let's implement that for Parquet as well.
```
  common::CompressionKind getCompression() const {
    return readerBase_->getCompressionKind();
  }
```

Reply:
Instead of returning the thrift file Metadata, I can create a function **getCompression()** and  return the compression Codec directly as **fileMetaData.row_groups[0].columns[0].meta_data.codec;**. This works, however as compression is stored on the column level, can different columns have different compression Kinds?

Reply:
If there is really a need later, we can extend the API to specify the row_group number and column number.